### PR TITLE
[Chainsaw Tests] Add Chainsaw test for Bare Pods

### PR DESCRIPTION
**Pull request by Jay179-sudo**
## Related Issue(s)
Partial fix for #950 

## Description
Added chainsaw test for bare pods. Also added a test pod and a cluster-role manifest file.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by Jay179-sudo** [comment]
Maybe we can also reduce the scheduled time on the ClusterCleanupPolicy file from five minutes down to one minute to reduce CI time?
Please do let me know if any other issues persist.

**Comment by realshuting** [comment]
> Maybe we can also reduce the scheduled time on the ClusterCleanupPolicy file from five minutes down to one minute to reduce CI time? Please do let me know if any other issues persist.

Makes sense, can you replicate the same policy and reduce the scheduled time to a minute?

File: cleanup/cleanup-bare-pods/cleanup-bare-pods.yaml [hunk 1]
```diff
@@ -28,4 +28,4 @@ spec:
     - key: "{{ target.metadata.ownerReferences[] || `[]` }}"
       operator: Equals
       value: []
-  schedule: "*/5 * * * *"
+  schedule: "*/1 * * * *"
```
Comment (realshuting) [inline cleanup/cleanup-bare-pods/cleanup-bare-pods.yaml:31]: Not sure if it's ok to update the original policy, I was thinking to replicate the policy in the test folder and update the schedule.

@chipzoller - is this policy used anywhere else?

**Review by realshuting — APPROVED** [review]

**Review by chipzoller — CHANGES_REQUESTED** [review]
Please move the ClusterRole inside the `.chainsaw-test` directory.

File: cleanup/cleanup-bare-pods/.chainsaw-test/cleanup-bare-pods-test.yaml
Comment (chipzoller) [inline cleanup/cleanup-bare-pods/.chainsaw-test/cleanup-bare-pods-test.yaml:1]: The policy should not be duplicated like this even if the need is to just modify a field. Please use an appropriate step in the Chainsaw test to modify the `schedule` field of the existing policy.

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: cleanup/cleanup-bare-pods/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  labels:
+    app.kubernetes.io/component: cleanup-controller
+    app.kubernetes.io/instance: kyverno
+    app.kubernetes.io/part-of: kyverno
+  name: kyverno:cleanup-controller:temp
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - "pods"
```
Comment (chipzoller) [inline cleanup/cleanup-bare-pods/clusterrole.yaml:14]:
```suggestion
  - pods
```

File: cleanup/cleanup-bare-pods/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  namespace: kyverno
+  labels:
+    app.kubernetes.io/component: cleanup-controller
+    app.kubernetes.io/instance: kyverno
+    app.kubernetes.io/part-of: kyverno
+  name: kyverno:cleanup-controller:temp
```
Comment (chipzoller) [inline cleanup/cleanup-bare-pods/clusterrole.yaml:9]:
```suggestion
  name: kyverno:cleanup-controller:barepods
```

**Comment by Jay179-sudo** [comment]
@chipzoller any updates on this?

File: cleanup/cleanup-bare-pods/.chainsaw-test/clusterrole.yaml
Comment (mohamedawnallah) [inline cleanup/cleanup-bare-pods/.chainsaw-test/clusterrole.yaml:1]: Rename this file to `cluster-role.yaml` to be consistent with other files in the codebase?

**Review by mohamedawnallah — APPROVED** [review]
Great work @Jay179-sudo! 💪 Left some nit comments...

File: cleanup/cleanup-bare-pods/.chainsaw-test/chainsaw-step-02-assert-1.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kyverno.io/v2beta1
+kind: ClusterCleanupPolicy
+metadata:
+  name: clean-bare-pods
```
[the remark this answers is not in the document]
Comment (Jay179-sudo) [reply cleanup/cleanup-bare-pods/.chainsaw-test/chainsaw-step-02-assert-1.yaml:4]: The status.ready field is not defined for a ClusterCleanupPolicy object. It is defined within status.conditions.

When I ran the tests as normal, the status field that does get filled up is the lastExecutionTime field. However, this is an field that only gets filled up after a round of execution which will not allow for direct condition checks in our chainsaw test file.

Wouldn't it be better to skip this field altogether?

Status Fields during execution

[the remark this answers is not in the document]
Comment (mohamedawnallah) [reply cleanup/cleanup-bare-pods/.chainsaw-test/chainsaw-step-02-assert-1.yaml:4]: Yes you are right 👍

**Review by mohamedawnallah — APPROVED** [review]

File: cleanup/cleanup-bare-pods/.chainsaw-test/chainsaw-test.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: cleanup-bare-pods
+spec:
+  steps:
+    - name: apply clusterrole
+      try:
+        - apply:
+            file: clusterrole.yaml
```
Comment (mohamedawnallah) [inline cleanup/cleanup-bare-pods/.chainsaw-test/chainsaw-test.yaml:10]: @Jay179-sudo It seems you forgot to rename this as well to `cluster-role.yaml`.

PS:
Also the DCO signing

**Review by chipzoller — APPROVED** [review]

**Pull request merged**